packages/inter-protocol/src/auction/scheduler.js

```diff
@@ -116,42 +117,103 @@ export const computeRoundTiming = (params, baseTime) => {
  * @property {() => void} startRound
  */
 
+/**
+ * @typedef {object} ScheduleNotification
+ *
+ * @property {Timestamp | undefined} activeStartTime start time of current
+ *    auction if auction is active
+ * @property {Timestamp} nextStartTime start time of next auction
+ * @property {Timestamp} nextDescendingStepTime when the next descending step
+ *    will take place
+ */
+
 /**
  * @param {AuctionDriver} auctionDriver
  * @param {import('@agoric/time/src/types').TimerService} timer
  * @param {Awaited<import('./params.js').AuctionParamManaager>} params
  * @param {import('@agoric/time/src/types').TimerBrand} timerBrand
+ * @param {Publisher<ScheduleNotification>} schedulePublisher
  */
 export const makeScheduler = async (
   auctionDriver,
   timer,
   params,
   timerBrand,
+  schedulePublisher,
 ) => {
   /**
    * live version is defined when an auction is active.
    *
    * @type {Schedule | undefined}
    */
   let liveSchedule;
-  /**
-   * Next should always be defined after initialization unless it's paused
-   *
-   * @type {Schedule | undefined}
-   */
-  let nextSchedule;
+
+  /** @returns {Promise<{now:Timestamp, nextSchedule: Schedule}>} */
+  const initializeNextSchedule = async () => {
+    return E.when(
+      // XXX manualTimer returns a bigint, not a timeRecord.
+      E(timer).getCurrentTimestamp(),
+      now => {
+        const nextSchedule = computeRoundTiming(
+          params,
+          TimeMath.toAbs(now, timerBrand),
+        );
+        return { now, nextSchedule };
+      },
+    );
+  };
+  let { now, nextSchedule } = await initializeNextSchedule();
+  const setTimeMonotonically = time => {
+    TimeMath.compareAbs(time, now) >= 0 || Fail`Time only moves forward`;
+    now = time;
+  };
+
   const stepCancelToken = makeCancelToken();
 
   /** @type {typeof AuctionState[keyof typeof AuctionState]} */
   let auctionState = AuctionState.WAITING;
 
+  const publishSchedule = () => {
+    const calcNextStep = () => {
```

Comment:
this function should move to the left so it's not redefined whenever `publicSchedule()`.

with less context it could help for the name to be more specific. e.g. `calcNextDescendingStepTime()`.

though consider moving it into another module as a function that takes `liveSchedule` and `nextSchedule`. Easier to test that way. Actually, _please_ move it out and have a unit test for it.